src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/RequestsForPlacementTest.kt

```diff
@@ -182,7 +190,11 @@ class RequestsForPlacementTest : IntegrationTestBase() {
             .expectBody()
             .json(
               objectMapper.writeValueAsString(
-                requestForPlacementTransformer.transformPlacementRequestEntityToApi(placementRequest),
+                requestForPlacementTransformer.transformPlacementRequestEntityToApi(
+                  placementRequest,
+                  // user created associated application, so they can withdraw the placement request
+                  canBeDirectlyWithdrawn = true,
+                ),
```

Comment:
hmm, maybe I'm missing it, but does it feel like this logic around the withdrawal rules should to be laid out in a unit test rather than in an integration test? i.e. that a class, e.g. `RequestForPlacementWithdrawal(rfp, user): boolean` should have responsibility for making this decision, or is it ok to have in the RfPTransformer?

Reply:
In this case because we're testing the entire JSON response i was forced to determine whether it was withdrawable or not, and it's not immediately obvious from the test data so I added a comment to explain why this is the case. I think we'll always have this challenge in integration tests where we're returning withdrawable status and test the whole JSON

As for where we test withdrawal logic, it's partially split across entity-specific service unit testing  (e.g. PlacementRequestService.getWithdrawableState), the Cas1WithdrawableTreeBuilderTest , and then WithdrawalTest which provides some high level integration tests and 'fills the gap' of testing where the state of one entity in the tree affects some other entity (e.g. booking blocks a placement request withdrawal)

On completing the PR I decided I didn't like passing the list of WithdrawableEntities into the transformer, especially as i've added a function in the transformer that takes a boolean in instead. I was going to change that in my next PR that calls the delius endpoint, but i think i'll change that now.